libtiledbsoma/src/pyapi/libtiledbsoma.cc

```diff
@@ -283,24 +283,110 @@ PYBIND11_MODULE(libtiledbsoma, m) {
             "batch_size"_a = "auto",
             "result_order"_a = "auto")
 
+        // The following short functions are expected to be invoked when the
+        // coords are Python list/tuple, or NumPy arrays.  Arrow arrays are in
+        // the long if-else-if function below.
+        //
         // Binding overloaded methods to templated member functions requires
         // more effort, see:
         // https://pybind11.readthedocs.io/en/stable/classes.html#overloaded-methods
+
+        // In an initial version of this file we had `set_dim_ranges` relying
+        // solely on type-overloading. This worked since we supported only int
+        // and string indices. In a subsequent version we are now supporting
+        // various NumPy/PyArrow types including float32, float64, int8, uint16,
+        // etc. It is an unfortunate fact that pybind11 does _not_ successfully
+        // disambiguate between float32 and float64, or between int8 and int64,
+        // etc. given that we ask it to disambiguate using not just types but
+        // std::vector of types or std::vector of std::pair of types.
+        // Experiments have shown that when both float32 and float64 are
+        // implemented with overloaded names to be differentiated solely by
+        // type, pybind11 uses the _first found_. Therefore it is necessary for
+        // us to no longer use common overloaded names.
+
         .def(
             "set_dim_points",
             static_cast<void (SOMAReader::*)(
-                const std::string&, const std::vector<int64_t>&)>(
+                const std::string&, const std::vector<std::string>&)>(
                 &SOMAReader::set_dim_points))
 
         .def(
             "set_dim_points",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<int64_t>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_string_or_bytes",
```

Comment:
This looks like the same as the `set_dim_points` with `std::vector<std::string>` above. Are they both needed for pybind11 to bind properly?

Reply:
@gspowley good call & i was thinking the same thing ... best to just "let go" of the plain `set_dim_points` everywhere ...